src/garage/torch/algos/sac.py

```diff
@@ -231,24 +237,32 @@ def _get_log_alpha(self, **kwargs):
         log_alpha = self.log_alpha
         return log_alpha
 
-    def _temperature_objective(self, log_pi):
+    def _temperature_objective(self, log_pi, get_log_alpha_kwargs=None):
         """Compute the temperature/alpha coefficient loss.
 
         Args:
             log_pi(torch.Tensor): log probability of actions that are sampled
                 from the replay buffer. Shape is (1, buffer_batch_size).
+            get_log_alpha_kwargs(dict): keyword args that can be used in
+                retrieving the log_alpha parameter.
 
         Returns:
             torch.Tensor: the temperature/alpha coefficient loss.
 
         """
+        if not get_log_alpha_kwargs:
+            get_log_alpha_kwargs = {}
         alpha_loss = 0
         if self.use_automatic_entropy_tuning:
-            alpha_loss = (-(self._get_log_alpha()) *
+            alpha_loss = (-(self._get_log_alpha(**get_log_alpha_kwargs)) *
                           (log_pi.detach() + self.target_entropy)).mean()
         return alpha_loss
 
-    def _actor_objective(self, obs, new_actions, log_pi_new_actions):
+    def _actor_objective(self,
+                         obs,
+                         new_actions,
+                         log_pi_new_actions,
+                         get_log_alpha_kwargs=None):
```

Comment:
what is the purpose of get_log_alpha_kwargs? this seems like something the function should be determining. how the hell would a caller know what the kwargs to a subroutine should be???